language/compiler/ir-to-bytecode/src/context.rs

```diff
@@ -252,7 +252,10 @@ impl<'a> Context<'a> {
             assert!(options[idx as usize].is_none());
             options[idx as usize] = Some(item);
         }
-        options.into_iter().map(|opt| opt.unwrap()).collect()
+        options
+            .into_iter()
+            .map(|opt| opt.expect("items produces size distinct index values"))
```

Comment:
MIRAI heuristically treats expect as an assume and unwrap as an assert. In this case we need to assume since the necessary precondition will involve a quantifier that is not currently supported.